src/components/FileUploadModal/upload-local-files.ts

```diff
@@ -15,104 +15,100 @@ import { VIEWER_ERROR } from "../../state/viewer/constants";
+files. The Antd Upload wraps the Upload component from react-component.
 - Antd docs: https://ant.design/components/upload/#API
 - rc-upload docs: https://github.com/react-component/upload#customrequest
-
-This custom request function takes in an array of all the files loaded by the
-user, finds and processes the .simularium file and any geometry files separately, 
-then tells the app to load the trajectory with its associated geometry files 
-into the viewer.
 */
-export default (
-    { onSuccess, onError }: UploadRequestOption,
+export default async (
     selectedFiles: File[],
     clearSimulariumFile: (isNewFile: { newFile: boolean }) => void,
     loadFunction: (simulariumFile: LocalSimFile) => void,
     setViewerStatus: (status: { status: ViewerStatus }) => void,
-    setError: (error: ViewerError) => void
+    setError: (error: ViewerError) => void,
+    rcRequest?: UploadRequestOption
 ) => {
-    numCustomRequests++;
-    if (numCustomRequests !== 1) {
+    numRequests++;
+    if (numRequests !== 1) {
         // If the user loads multiple files at once (.simularium file + geometry file(s)),
         // this function is called multiple times, but we only need to process
         // and load the trajectory once.
-        if (numCustomRequests === selectedFiles.length) {
+        if (numRequests === selectedFiles.length) {
             // We're at the last customRequest for this batch of files,
             // so reset counter
-            numCustomRequests = 0;
+            numRequests = 0;
         }
         return;
     }
 
     if (selectedFiles.length === 1) {
         // numCustomRequests and selectedFiles.length are both 1, so reset
-        numCustomRequests = 0;
+        numRequests = 0;
     }
 
     // want the loading indicator to show without any lag time
     // as soon as user hits "Open" button or drops files,
     // and not have to have this action called multiple places in the code.
     clearSimulariumFile({ newFile: true });
 
-    const filesArr: File[] = Array.from(selectedFiles) as File[];
-
     try {
         // Try to identify the simularium file.
         // Put all the other files as text based geoAssets.
-        const simulariumFileIndex = findIndex(filesArr, (file) =>
+        const simulariumFileIndex = findIndex(selectedFiles, (file) =>
             file.name.includes(".simularium")
         );
         if (simulariumFileIndex === -1) {
             throw new Error(
                 "Trajectory file was not found; please make sure it has a .simularium extension."
             );
         }
-        Promise.all<string | ISimulariumFile>(
-            filesArr.map((element, index) => {
+        const parsedFiles = await Promise.all<string | ISimulariumFile>(
+            selectedFiles.map((element, index) => {
                 if (index !== simulariumFileIndex) {
                     // is async call
                     return element.text();
```

Comment:
This is kinda out of scope of this ticket, but do you know if there are limitations to how the max file size that `.text()` can handle? We do a similar thing in the simularium-viewer call and anecdotally it seemed to fail loading the file somewhere in the ~100MB range, which is unfortunately within the size range of what input files may be. I think it'd be out of scope for this ticket to address that, but it probably is within scope to ensure we're handling any errors that may come out of this

Reply:
I didn't make a lot of substantive change to this function, so I don't think I considered this. It looks like the `try`/`catch` this is enclosed in ought to handle the case where the conversion errors out, though that doesn't keep it safe from freezes/crashes. I wasn't able to turn up a specified reason on Google for why the memory ceiling would be so low - for instance, strings [ought to work up to ~2GB](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#description) on reasonable systems.